src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java

```diff
@@ -230,7 +233,10 @@ public static Token parse(String sourceToken, JsonObject claims) throws TokenVal
    * @throws TokenValidationException Throws this when any validation step fails, but may throw
    * other exceptions as well.
    */
-  protected void validateCommon(HttpServerRequest request) throws TokenValidationException {
+  protected Future<Token> validateCommon(TokenValidationContext context) {
```

Comment:
Is `isCrossTenantRequest` really in scope for _every_ token type or is it _only_ in scope for `LegacyAccessToken` and `AccessToken`? Although it might appear to work even if you have this check being made in every request, I'm not sure if it is "correct" for the other token types, especially `DummyToken` and the newer token types that aren't released yet.

The idea with `validateCommon` is that only the items that are shared by every token type be placed here. 

The idea with these token type classes is that each type have responsibility performing the validation of its type in the implementation of the `validate` method. If the two access token types (`LegacyAccessToken` and the newer unreleased `AccessToken`) share validation logic perhaps create a base abstract class that these two can inherit from with that shared logic.

Reply:
ModuleToken is in this list as well, DummyToken probably not

Reply:
I think checking is tenant mismatch - it is core check the same as checking id userId is matched that already contained in validateCommon method [Link](https://github.com/folio-org/mod-authtoken/blob/master/src/main/java/org/folio/auth/authtokenmodule/tokens/Token.java#L265) , and if we remain userId validation in common method and move token validation to subclasses - it will look inconsistent as for me.
Also when adding new token type author can forget about token mismatch validation that can lead to some security hall, because tenant check - it is core validation check

Reply:
And also before this PR tenant mismatch was checking for all token types, because it was located in validateCommon, and if we are going to move this logic to only particular subclasses - we will not be safe to make this change , because we will broke existing validation contracts for exisitng token types

Reply:
Every token type implementation has an abstract `validateContext` method which is called by `Token.validate`. This is where logic that is unique to a given token type belongs. Since `validateContext` is called for every request where a token is in scope, putting validation logic there is "safe". Only if logic is shared by _every_ token type it should go in `validateCommon`. Every `Token` implementation is responsible for its own validation through its `validateContext` implementation.

We shouldn't put this new code in `validateCommon` because it convenient but because it is correct. To me, if this new check of the users table isn't needed by every token type, putting it in `validateCommon` is what could break the contract.

Do I know for a fact that this new logic isn't needed by every token type? I don't, but I suspect that they don't. I think being sure about this should be part of this change. Its easy enough to see the lifecycle of each token type. Ask yourself, will your ECR changes require the user table check for each type type? If you're not completely sure, its safer to change only the token types you are sure of (the access tokens?), and then see what breaks by deploying mod-authtoken in a full system, rather than changing every token type "just in case".

Reply:
okay, we will rewrite this

Reply:
@steveellis please review latest changes. thank you

Reply:
Ok giving token types the ability to opt out of the logic with the override seems fine. It looks like `ModuleToken` and `DummyToken` will need the new logic based on how they are being used.

Reply:
yes, basically `ModuleToken` and `DummyToken ` should be cross-tenant as well